aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -242,6 +257,23 @@ class DispatchTable final {
         return first_tensor_arg.unsafeToTensorImpl()->type_id();
       }
     }
+
+    bool get_is_variable(const Stack* stack, const std::string& operator_name) const {
+      const IValue& first_tensor_arg = torch::jit::peek(
```

Comment:
this is a lot of code duplication. Can we refactor this into a `TensorImpl* getDispatchArgument(Stack* stack)` method that is used by both, `get_dispatch_key` and `get_is_variable`?

Also, now every operator call checks twice if the argument is a tensor list, once in `get_dispatch_key` and once in `get_is_variable`. This is a potential perf issue. Is there a way to only check it once and get both, the dispatch key and the variable wrapper from it?